src/Symfony/Component/Yaml/Inline.php

```diff
@@ -104,6 +108,18 @@ public static function dump($value, $exceptionOnInvalidType = false, $objectSupp
 
                 return 'null';
             case is_object($value):
+                if ($value instanceof \DateTime) {
```

Comment:
So actually we should check for both concrete `DateTimeInterface` implementations:

```php
if ($value instanceof \DateTime || $value instanceof \DateTimeImmutable) {
    // ...
}
```